src/client/encode.rs

```diff
@@ -1,73 +1,16 @@
-//! Process HTTP connections on the client.
-
-use async_std::io::{self, BufReader, Read, Write};
+use async_std::io::{self, Read};
 use async_std::prelude::*;
 use async_std::task::{Context, Poll};
-use http_types::{ensure, ensure_eq, format_err};
-use http_types::{
-    headers::{HeaderName, HeaderValue, CONTENT_LENGTH, DATE, TRANSFER_ENCODING},
-    Body, Request, Response, StatusCode,
-};
+use http_types::format_err;
+use http_types::Request;
 
-use std::convert::TryFrom;
 use std::pin::Pin;
-use std::str::FromStr;
 
-use crate::chunked::ChunkedDecoder;
 use crate::date::fmt_http_date;
-use crate::{MAX_HEADERS, MAX_HEAD_LENGTH};
-
-/// An HTTP encoder.
-#[doc(hidden)]
-#[derive(Debug)]
-pub struct Encoder {
-    /// Keep track how far we've indexed into the headers + body.
-    cursor: usize,
-    /// HTTP headers to be sent.
-    headers: Vec<u8>,
-    /// Check whether we're done sending headers.
-    headers_done: bool,
-    /// Request with the HTTP body to be sent.
-    request: Request,
-    /// Check whether we're done with the body.
-    body_done: bool,
-    /// Keep track of how many bytes have been read from the body stream.
-    body_bytes_read: usize,
-}
-
-impl Encoder {
-    /// Create a new instance.
-    pub(crate) fn new(headers: Vec<u8>, request: Request) -> Self {
-        Self {
-            request,
-            headers,
-            cursor: 0,
-            headers_done: false,
-            body_done: false,
-            body_bytes_read: 0,
-        }
-    }
-}
-
-/// Opens an HTTP/1.1 connection to a remote host.
-pub async fn connect<RW>(mut stream: RW, req: Request) -> http_types::Result<Response>
-where
-    RW: Read + Write + Send + Sync + Unpin + 'static,
-{
-    let mut req = encode(req).await?;
-    log::trace!("> {:?}", &req);
-
-    io::copy(&mut req, &mut stream).await?;
-
-    let res = decode(stream).await?;
-    log::trace!("< {:?}", &res);
-
-    Ok(res)
-}
 
 /// Encode an HTTP request on the client.
 #[doc(hidden)]
-async fn encode(req: Request) -> http_types::Result<Encoder> {
+pub async fn encode(req: Request) -> http_types::Result<Encoder> {
```

Comment:
This probably should be called `encoder` since it doesn't actually encode the request.